action.js

```diff
@@ -1,41 +1,100 @@
-if (localStorage.getItem('book') == null) {
-  localStorage.setItem('book', JSON.stringify([]));
+class Inputdata {
+  constructor(intitle, inauthor) {
+    this.intitle = intitle;
+    this.inauthor = inauthor;
+  }
 }
+class Output {
+  /* eslint-disable class-methods-use-this */
+  create() {
+    if (localStorage.getItem('books') == null) {
+      localStorage.setItem('books', JSON.stringify([]));
+    }
+    return JSON.parse(localStorage.getItem('books'));
+  }
 
-const totalBooks = JSON.parse(localStorage.getItem('book'));
-const container = document.querySelector('.extra');
-const main = document.createElement('div');
-main.className = 'main';
-
-function add() {
-  if (document.querySelector('.title').value !== '' || document.querySelector('.author').value !== '') {
-    const book = {
-      title: document.querySelector('.title').value,
-      author: document.querySelector('.author').value,
-    };
-    totalBooks.push(book);
-    localStorage.setItem('book', JSON.stringify(totalBooks));
+  /* eslint-enable class-methods-use-this */
+  add() {
+    const totalbooks = this.create();
+    const data = new Inputdata(document.querySelector('.title').value, document.querySelector('.author').value);
+    if (document.querySelector('.title').value !== '' && document.querySelector('.author').value !== '') {
+      const book = {
+        title: data.intitle,
+        author: data.inauthor,
+      };
+      totalbooks.push(book);
+      localStorage.setItem('books', JSON.stringify(totalbooks));
+    }
   }
-}
 
-function showBook() {
-  if (totalBooks.length > 0) {
-    for (let i = 0; i < totalBooks.length; i += 1) {
-      main.innerHTML += `<ul><li>
-      ${totalBooks[i].title}<br>
-      ${totalBooks[i].author}<br>
-     <button onclick="remove(${i})" class='remove'>Remove</button><br></li></ul>
-    `;
-      container.append(main);
+  show() {
+    const totalBooks = this.create();
+    const container = document.querySelector('.extra');
+    const main = document.createElement('section');
+    main.className = 'main';
+    if (totalBooks.length > 0) {
+      for (let i = 0; i < totalBooks.length; i += 1) {
+        main.innerHTML += `<section class='outer'><section class='inner'><p>
+           " ${totalBooks[i].title} " by
+            ${totalBooks[i].author}</p>
+         <p><button class='removeitem'>Remove</button></p></section></section>
+          `;
+        container.appendChild(main);
+      }
     }
   }
+
+  remove(e) {
+    const totalBooks = this.create();
+    totalBooks.splice(e, 1);
+    localStorage.setItem('books', JSON.stringify(totalBooks));
+    document.querySelector('.inner').innerHTML = '';
```

Comment:
- [ ] [OPTIONAL] I have noticed that when deleting the books, the first book in array is the one deleted instead of targeting the specic id clicked on. Kindly ensure that the removing targets the book needed to be deleted ☺️.